### PR TITLE
[SwitchListTile] adds controlAffinity property

### LINKED ISSUE #22134: SwitchListTile has no controlAffinity property like CheckboxListTile or RadioListTile
Why does SwitchListTile has no **controlAffinity** property like CheckboxListTile or RadioListTile?

Please add, thanks!

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Adds `controlAffinity` property to change the position of `Switch`.

## Related Issues
Fixes #22134 

## Tests

I added the following tests:
- Added test to check  `controlAffinity` property works as desired and position of `Switch` is changed.
- Added test to check the default value of `controlAffinity` works as desired.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by shihaohong — APPROVED** [review]
LGTM

**Comment by shihaohong** [comment]
Looks like the analyzer is complaining about trailing whitespace

**Comment by shihaohong** [comment]
Btw, I just realized you're not listed [as an AUTHOR](https://github.com/flutter/flutter/blob/master/AUTHORS) on the AUTHORS page. If you want, feel free to add yourself on there since you've been contributing a great deal in the last few months

**Comment by AyushBherwani1998** [comment]
> 
> 
> Looks like the analyzer is complaining about trailing whitespace

I ran the analyzer in the test folder and it didn't find any issue. Line 301 mentioned is a new line.

**Comment by AyushBherwani1998** [comment]
> 
> 
> Btw, I just realized you're not listed [as an AUTHOR](https://github.com/flutter/flutter/blob/master/AUTHORS) on the AUTHORS page. If you want, feel free to add yourself on there since you've been contributing a great deal in the last few months

Thank you 😄 
I have updated the AUTHORS

File: AUTHORS [hunk 1]
```diff
@@ -56,3 +56,4 @@ Michael Lee <ckmichael8@gmail.com>
 Katarina Sheremet <katarina@sheremet.ch>
 Nicolas Schneider <nioncode+git@gmail.com>
 Mikhail Zotyev <mbixjkee1392@gmail.com>
+Ayush Bherwani <ayush.bherwani1998@gmail.com>
```
Comment (shihaohong) [inline AUTHORS:59]: nit: add a newline to the end of files
```suggestion
Ayush Bherwani <ayush.bherwani1998@gmail.com>

```

**Review by shihaohong — APPROVED** [review]

File: packages/flutter/test/material/switch_list_tile_test.dart [hunk 2]
```diff
@@ -298,4 +298,44 @@ void main() {
     await tester.pump();
     expect(Focus.of(childKey.currentContext, nullOk: true).hasPrimaryFocus, isFalse);
   });
+
```
Comment (shihaohong) [inline packages/flutter/test/material/switch_list_tile_test.dart:301]:
```suggestion

```

Comment (AyushBherwani1998) [reply packages/flutter/test/material/switch_list_tile_test.dart:301]: But, isn't that a new line? I mean, in previous PRs it never gave this error.

Comment (shihaohong) [reply packages/flutter/test/material/switch_list_tile_test.dart:301]: It's not the new line. It's the whitespace within the line. 
```
{whitespace}{whitespace}
```
vs
```

```

Comment (AyushBherwani1998) [reply packages/flutter/test/material/switch_list_tile_test.dart:301]: I see, thanks for clearing silly doubt @shihaohong.
I'll do the required changes.

**Pull request merged**